extension/app/ts/components/pages/Home.tsx

```diff
@@ -59,31 +59,38 @@ function FirstCard(param: FirstCardParams) {
 			{ param.useSignersAddressAsActiveAddress || !param.simulationMode ?
 				<p style = 'color: var(--text-color); text-align: left'>
 					<span class = 'vertical-center' style = 'display: inline-flex;' >
-						Retrieving from&nbsp;
-						<SignersLogoName signerName = { param.signerName } />
+						{ param.signerName === 'NoSigner' ? 'Trying to retrieve from a signer, but signer is not connected' :
+							<>Retrieving from&nbsp;
+								<SignersLogoName signerName = { param.signerName } />
+							</>
+						}
 					</span>
 					{ param.signerAccounts !== undefined && param.signerAccounts.length > 0 && param.tabIconDetails.icon !== ICON_NOT_ACTIVE ? <span style = 'float: right; color: var(--primary-color);'> CONNECTED </span> :
 						param.tabIconDetails.icon === ICON_SIGNING || param.tabIconDetails.icon === ICON_SIGNING_NOT_SUPPORTED ? <span style = 'float: right; color: var(--negative-color);'> NOT CONNECTED </span> : <></>
 					}
 				</p>
 				: <></>
 			}
-			{ param.activeAddress !== undefined ?
-				<ActiveAddress
-					activeAddress = { {
-						type: 'addressInfo' as const,
-						...param.activeAddress,
-					} }
-					buttonText = { 'Change' }
-					disableButton = { !param.simulationMode }
-					changeActiveAddress = { param.changeActiveAddress }
-					renameAddressCallBack = { param.renameAddressCallBack }
-				/>
-			: param.useSignersAddressAsActiveAddress || !param.simulationMode ?
-				<div class = 'content' style = 'color: var(--negative-color)'>
-					{ `No active address found in ${ getPrettySignerName(param.signerName) }` }
-				</div>
-			:
+
+			<ActiveAddress
+				activeAddress = { param.activeAddress !== undefined ? {
+					type: 'addressInfo' as const,
+					...param.activeAddress,
+				} : undefined }
```

Comment:
I struggled a bit to mentally parse this.  I eventually got it but I think it may actually help to one-line this so it "looks" similar to the other lines below it (which would have helped my brain deduce what was happening.  Alternatively, having the ternary with `?` and `:` on the start of lines may have helped because that is what I'm used to for multiline ternaries.
```suggestion
				activeAddress = { param.activeAddress !== undefined ? { type: 'addressInfo' as const, ...param.activeAddress } : undefined }
```